compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -303,6 +303,15 @@ pub enum StatementKind<'tcx> {
     /// and modify the place in an opaque way.
     Retag(RetagKind, Box<Place<'tcx>>),
 
+    /// This statement exists to preserve a trace of a scrutinee matched against a wildcard
+    /// binding.  This is especially useful for `let _ = PLACE;` bindings that desugar to a single
+    /// `PlaceMention(PLACE)`.
+    ///
+    /// When executed at runtime this is a nop.
+    ///
+    /// Disallowed after drop elaboration.
```

Comment:
Why is that?

Reply:
Simply because we remove it after drop elaboration. There is no particular reason to ban it either, except consistency.

Reply:
Okay. This needs to be removed with #104844 then, since in Miri it clearly sticks around until after drop elaboration.